third/lprng/lprng/man/lp.1
.ds VE LPRng-3.5.3
'\" e
.\" @(#)lp.1 1.36 95/10/09 SMI; from SVr4
.\" Copyright 1989 AT&T
.\" Copyright (c) 1995, Sun Microsystems, Inc.
.\" All Rights Reserved
.\" Portions Copyright (c) 1992, X/Open Company Limited
.\" Portions Copyright (c) 1996, Patrick Powell
.\" All Rights Reserved
.ig
.EQ
gsize 10
delim $$
.EN
..
.TH LP 1 \*(VE "LPRng"
.SH NAME
lp \- send requests to an LPRng print service
.SH SYNOPSIS
.B lp
[
.B \-A
] [
.B \-c
] [
.B \-m
] [
.B \-p
] [
.B \-s
] [
.B \-w
] [
.BI \-d " dest"
]
.if n .ti +5n
[
.BI \-f " form-name"
[
.B \-d any
] ]
.if t .ti +5n
[
.BI \-H " special-handling"
]
.if n .ti +5n
[
.BI \-n " number"
] [
.BI \-o " option"
] [
.BI \-P " page-list"
]
.if n .ti +5n
[
.BI \-q " priority-level"
]
.if t .ti +5n
[
.BI \-S " character-set"
[
.B \-d any
] ]
.if n .ti +5n
[
.BI \-S " print-wheel"
[
.BI \-t " title"
]
.if t .ti +5n
.if n .ti +5n
[
.BI \-T " content-type"
[
.B \-r
] ] [
.BI \-y " mode-list"
]
.if n .ti +5n
[
.BI \-D " debug-options"
]
.if n .ti +5n
[
.IR file \|.\|.\|.
]
.SH DESCRIPTION
.IX "lp command" "" "\fLlp\fP \(em send requests to an LP print service"
.IX "LPRng print services" "print files" "" "print files \(em \fLlp\fP"
.IX "printers" "send requests" "" "send requests \(em \fLlp\fP"
.LP
The LPRng
\f3lpr\fP
client supports an \f3lp\fP interface when invoked with the
\f3lp\fP
program name.
This is usually done by making a symbolic link to the
\f3lpr\fP program with the name
\f3lp\fP.
This is similar to the
.BR ex (1)
and
.BR vi (1)
programs.
The supported form of the \f3lp\fP command
arranges for the named \f2file(s)\fP and associated information
(collectively called a
.IR request )
to be printed.
If no file names are specified on the command line,
the standard input is assumed.
The standard input may be specified
along with a named \f2file\f1(s)
on the command line by listing the file name(s)
and specifying \`\f3\-\f1\' (dash)
for the standard input.
The \f2files\f1 will be printed
in the order in which they appear on the shell command line.
.LP
The
.SM LP
print service
associates a unique \f2request-ID\f1
with each request
and displays it on the standard output.
The job number portion or full request-ID can be used later with
.BR cancel (1)
or
.BR lprm (1)
when canceling a job or when determining its status.
See 
.BR cancel (1)
and
.BR lprm (1)
for details about canceling a request,
and
.BR lpstat (1)
and
.BR lpq (1)
for information about checking the status of a print request.
.SH OPTIONS
.LP
Options to
.B lp
always precede any file names,
but may be specified in any order.
The following options are available for
.BR lp :
.TP 15
.B \-A
Use authenticated transfer.
Note that this is not a standard
.BR lp (1)
option but has been added for compatibilty with LPRng.
.TP 15
.B \-c
Make a copy
of the
.I file
before printing.
(This has no effect as LPRng always sends
jobs immediately to the remote server for printing.)
.TP
.BI \-d " dest"
Choose
.I dest
as the printer or class of printers
that is to do the printing.
By default,
.I dest
is taken from
the environment variable
.B \s-1LPDEST\s+1
(if it is set),
and then 
.B \s-1PRINTER\s+1 .
Otherwise, a default destination (if one exists)
for the computer system is used.
.TP
\f3\-f \f2form-name\f1
Print the request on the form
.IR form-name .
The first letter of the
\f2form-name\f1
option is used as the LPRng filter format.
This is similar to the lpr -F option.
.TP 15
.BI \-H " special-handling"
These options are ignored.
.TP 15
.B \-m
Send mail (see
.BR mail (1))
after the files have been printed.
The mail address is taken from the USER enviroment
variable.
By default, no mail is sent
upon normal completion of the print request.
.TP
.BI \-n " number"
Print
.I number
copies (default is
.BR 1 )
of the output.
.br
.ne 5
.TP
.BI \-o " option"
Specify printer-dependent
.IR options .
Several such
.I options
may be collected by specifying
the
.B \-o
keyletter more than once
(\f3\-o $option sub 1$\f3
\-o $option sub 2$\f2 ... \f3\-o $option sub n$\f1),
or by specifying
the \f3\-o\f1 keyletter followed by
a list of options
enclosed in double quotes
(that is, \f3\-o "$option sub 1$\f3
$option sub 2$\f3 \f2... $option sub n$\f3"\f1).
The following options have a immediate effect;
other options are passed as a
.B Z
line entry in the control file,
i.e. - as an 
.BI lpr (1)
.B \-Z
option.
.RS 15
.TP 10
.B nobanner
Do not print a banner page with this request.
This is the same as the
.BI lpr (1)
.B -h
(no header) option.
.TP  10
\f3width\f2=number\f1
Print this request with page-width set to
\f2number\f1.
This is the same as the
.BI lpr (1)
.BI -w " width"
option.
.RE
.TP 15
.BI \-P " page-list"
Print the pages specified in \f2page-list\f1.
This option is ignored.
.TP
.B \-p
Enable notification on completion of the print request.
This option is ignored.
.TP
.BI \-q " priority-level"
Assign this request \f2priority-level\f1 in the printing queue.
The values of \f2priority-level\f1 range from
.BR 0 ,
the highest priority, to
.BR 25 ,
the lowest priority.  If a priority is not specified, the default
for the print service is used, as assigned by the system administrator.
A priority limit may be assigned to individual users by the
system administrator.
The numerical values are translated to corresponding letter
priority values.
.TP 15
.B \-s
Suppress messages from \f3lp\f1
such as those that begin with "\f3request id is\f1\|.\|.\|."
.PD 0
.TP 15
\f3\-S \f2character-set \f1
.TP 15
\f3\-S \f2print-wheel\f1
.PD
Print this request using the specified \f2character-set\f1
or \f2print-wheel\f1.
The options are passed as an
.BI lpr (1)
.B \-Z
option.
.br
.ne 7
.TP
\f3\-t \f2title\f1
Print
.I title
on the banner page of the output.
If
.I title
is not supplied the name of the file is printed on the banner
page.
Enclose \f2title\fP in quotes if it contains blanks.
.TP
\f3\-T \f2content-type\f1 [\f3\-r\f1]
Print the request on a printer that can support
the specified \f2content-type\f1.
The
\f2content-type\f1
is passed as an
.BI lpr (1)
.B \-Z
option.
.TP
\f3\-w\f1
Write a message on the user's terminal after the
.I files
have been printed.
This option is ignored.
.TP
\f3\-y \f2mode-list\f1
Print this request
according to the printing modes listed in \f2mode-list\f1.
The
\f2mode-list-type\f1
is passed as an
.BI lpr (1)
.B \-Z
option.
.TP
\f3\-D \f2debug-optios\f1
A list of debug options for the program.
These can be a number (debug level) or keyword and level (flag),
such as 9, network, network+2, etc.
.ne 15
.SH OPERANDS
The following operands are supported by
.BR lp :
.TP 15
.I file
A path name of a file to be output.
If no
.I file
operands are specified,
or if a
.I file
operand is
.BR \- ,
the standard input will be used.
The
.BR \-
option must be last in the list.
.SH ENVIRONMENT
See
.BR environ (5)
for descriptions of the following environment variables that affect the
execution of
.BR lp
and
.BR cancel :
.TP 15
.SB LPDEST
Determine the output device or destination.
If the
.SB LPDEST
environment variable is not set,
the
.SB PRINTER
environment variable will be used.
The
.BI \-d \0dest
option takes precedence over
.SB LPDEST\s0\f1.
Results are undefined when
.B \-d
is not specified and
.SB LPDEST
contains a value that is not a valid device or destination name.
.TP
.SB PRINTER
Determine the output device or destination.
If the
.SB LPDEST
and
.SB PRINTER
environment variables are not set, an unspecified output device is used.  The
.BI \-d \0dest
option and the
.SB LPDEST
environment variable takes precedence over
.SB PRINTER\s0\f1.
Results are undefined when
.B \-d
is not specified,
.SB LPDEST
is unset, and
.SB PRINTER
contains a value that is not a valid device or destination name.
.SH "EXIT STATUS"
The following exit values are returned by
.BR lp :
.TP 15
.B 0
All input files were processed successfully.
.TP
.B >0
No output device was available, or an error occurred.
.SH SEE ALSO
lpd.conf(5),
lpc(8),
lpd(8),
lpr(1),
lpq(1),
lprm(1),
printcap(5),
lpd.perms(5),
pr(1).
.SH DIAGNOSTICS
.nf
Most of the diagnostics are self explanatory.
If you are puzzled over the exact cause of failure,
set the debugging level on (-D5) and run again.
The debugging information will 
help you to pinpoint the exact cause of failure.
.fi
.SH "HISTORY"
.LP
LPRng is a enhanced printer spooler system
with functionality similar to the Berkeley LPR software.
In 1988 Patrick Powell released
the PLP (Public Line Printer) software,
which went through several evolutions.
Justin Mason (jmason@iona.ie)
generated PLP4.0 from several older releases of PLP.
In 1992 Patrick Powell
release LPRng,
a completely redesigned and newly written version of the software.
.LP
The LPRng mailing list is plp@iona.ie;
subscribe by sending mail to plp-request@iona.ie with
the word subscribe in the body.
The software is available from ftp://iona.ie/pub/LPRng.
.LP
LPRng is distributed under the GNU software license for non-commercial
use,
the Artistic License for limited commercial use. 
Commerical support and licensing is available through
Patrick Powell <papowell@sdsu.edu>.


doc/news
This file is intended to keep track of user-visible changes to the
release, for the construction of the Athena release notes.

Changes from release 8.0 to release 8.1:

	* Athena version numbers are now dotted triplets, e.g. 8.1.0
	  or 8.1.12.  Paths which include all or part of the Athena
	  version number now take the form "pathname-version",
	  e.g. "srvd-8.1" or "dash.menus-8.1"

	* In release 8.0, running "xmkmf" with the default path would
	  generate a Makefile which looked for its include files and
	  libraries in /mit/x11, which could be one of several
	  lockers.  Now it is preferrable to use the vendor X
	  libraries, so this bit of glue is going away.  There will be
	  no /usr/athena/bin/xmkmf any more, nor any symlinks in
	  /usr/athena pointing to the x11 locker.

	  Users can still do something like "add -f x11r5" and then
	  "xmkmf", and things should work, if the x11r5 locker is
	  built for their platform.

	* Athena bitmaps have moved to /usr/athena/include/X11/bitmaps.
	  Previously, they were in /usr/athena/lib/X11/bitmaps before
	  because of the /mit/x11 glue.

	* The symlink /bin/csh -> /bin/athena/tcsh has gone away.  The
	  update to 8.0 will fix /bin/csh references in /etc/passwd
	  and /etc/passwd.local to refer to /bin/athena/tcsh instead.

	* emacs18 has gone away.

	* AFS includes and libraries are installed in
	  /usr/afsws/include and /usr/afsws/lib.

	* wcl includes and libraries are installed in /usr/athena.

	* The basic krb5 headers and libraries are installed in
	  /usr/athena.

	* emacs has been upgraded from 19.30 to 19.34.  There should
	  be no significant user-visible changes.

	* RCS has been upgraded from version 4 to version 5.7.

	* x3270 has been upgraded from version 1.2 to version 3.1.1.3.

	* sysinfo is now available under Irix.

	* /bin/athena/tcsh now has Hesiod support enabled, which means
	  that "~lockername" will work in a pathname in most cases.
	  (It won't work for users with local passwd information who
	  don't have their lockers attached.)

	* For improved ease of debugging, many binaries are installed
	  with symbol tables, and compiled without optimization.
	  Therefore, the contents of /bin/athena and /usr/athena/bin
	  may be somewhat larger, and the binaries may consume more
	  CPU cycles, although it's unlikely that anyone will notice
	  the speed difference.


packs/glue/os/irix/lp.sh
#!/bin/sh
# $Id: lp.sh,v 1.1.6.1 2000-09-23 19:34:07 ghudson Exp $

# This script emulates the System V lp command using the Athena lpr
# command.  The emulation is not perfect; the known imperfections are:
#
# 	* The -s option is ignored, since lpr doesn't normally print
#	  output and doesn't have an option to suppress it when it
#	  does.
#
#	* The -m and -w options, which cause mail to be sent or a
#	  message to be written to the user when the print job is
#	  done, are both replaced with -z, which causes a zephyr to be
#	  sent to the user when the print job is done.
#
#	* All -o options are ignored (normally they are
#	  printer-specific options) except for "nobanner".
#
#	* Arguments to some options may be expanded twice for words and
#	  globbing.

# Parse command-line flags.
opts=""
title=""
suppress_s=no
usage="lp [-c] [-d dest] [-m] [-n number] [-t title] [-w] files"
while getopts cd:mn:o:st:w opt; do
	case "$opt" in
	c)
		suppress_s=yes
		;;
	d)
		opts="$opts -P $OPTARG"
		;;
	m)
		opts="$opts -z"
		;;
	n)
		opts="$opts -#$OPTARG"
		;;
	o)
		case "$OPTARG" in
		nobanner|*,nobanner|nobanner,*|*,nobanner,*)
			opts="$opts -h"
			;;
		esac
		;;
	s)
		;;
	t)
		title="$OPTARG"
		;;
	w)
		opts="$opts -z"
		;;
	\?)
		echo "$usage" 1>&2
		exit 1
		;;
	esac
done
if [ "$suppress_s" = no ]; then
	opts="$opts -s"
fi

shift `expr $OPTIND - 1`

if [ -z "$title" ]; then
	title="${@:-(stdin)}"
fi

exec /usr/athena/bin/lpr -J "$title" $opts "$@"
